proto/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use std::{env, path::PathBuf};
+
+use tonic_build::FileDescriptorSet;
+
+const RPC_PROTO: &str = "rpc.proto";
+
+#[cfg(feature = "internal")]
+const STORE_PROTO: &str = "store.proto";
+
+#[cfg(feature = "internal")]
+const BLOCK_PRODUCER_PROTO: &str = "block_producer.proto";
+
+pub fn rpc_file_descriptor() -> Result<FileDescriptorSet, protox::Error> {
+    let proto_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("proto");
+    protox::compile([RPC_PROTO], &[proto_dir])
+}
```

Comment:
It seems like here we compile the files again (similar to what we've done in the `build.rs` file. I thought the intent was to compile the files in `build.rs` and then just deserialize them here. In this scenario, we could do the following:

- Compile the protobuf files to their descriptors for each component separately in `build.rs`.
- Write these descriptors to disk.
- Here, just read the descriptors from disk using something like `include_bytes!`, decode and return them.